yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1054,11 +1084,7 @@ object Client extends Logging {
     System.setProperty("SPARK_YARN_MODE", "true")
     val sparkConf = new SparkConf
 
-    val args = new ClientArguments(argStrings, sparkConf)
-    // to maintain backwards-compatibility
-    if (!Utils.isDynamicAllocationEnabled(sparkConf)) {
-      sparkConf.setIfMissing(EXECUTOR_INSTANCES, args.numExecutors)
```

Comment:
I think this is handled in SparkSubmit now so isn't needed since no one should be using Client interface directly